packages/web/graphql/mutations/updateUsername.ts

```diff
@@ -1,12 +1,10 @@
 import gql from 'fake-tag';
 
 export const UpdateUsernameMutation = gql`
-  mutation UpdatePlayerUsername($username: String!) {
-    update_player(_set: { username: $username }, where: {}) {
-      returning {
-        id
-        username
-      }
+  mutation UpdatePlayerUsername($playerId: uuid!, $username: String!) {
```

Comment:
what error was the previous mutation causing ?

The backend is setup that only the player can update the player data information, which is checked from the header `X-Hasura-User-Id`

In case the backend is insecure => that is, if one player is able to update another players' data, then we need to update the permissions on the backend.

Reply:
It didn't work at all and just failed silently. I suspect it's because there is that hasura permission you mentioned. I tried to update another player's username in my testing and its not possible.